{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1d015d4e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Compiling model...\n",
      "Warning: For the backend tensorflow.compat.v1, `external_trainable_variables` is ignored, and all trainable ``tf.Variable`` objects are automatically collected.\n",
      "Building feed-forward neural network...\n",
      "'build' took 0.057357 s\n",
      "\n",
      "'compile' took 2.813442 s\n",
      "\n",
      "Initializing variables...\n",
      "Training model...\n",
      "\n"
     ]
    },
    {
     "ename": "InvalidArgumentError",
     "evalue": "Graph execution error:\n\nDetected at node 'sub_282' defined at (most recent call last):\n    File \"D:\\anaconda\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"D:\\anaconda\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel_launcher.py\", line 16, in <module>\n      app.launch_new_instance()\n    File \"D:\\anaconda\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 677, in start\n      self.io_loop.start()\n    File \"D:\\anaconda\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"D:\\anaconda\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 471, in dispatch_queue\n      await self.process_one()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 460, in process_one\n      await dispatch(*args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 367, in dispatch_shell\n      await result\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 662, in execute_request\n      reply_content = await reply_content\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 360, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 532, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2863, in run_cell\n      result = self._run_cell(\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2909, in _run_cell\n      return runner(coro)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n      coro.send(None)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3106, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3309, in run_ast_nodes\n      if await self.run_code(code, result, async_=asy):\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3369, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\86183\\AppData\\Local\\Temp\\ipykernel_19072\\2949448217.py\", line 203, in <cell line: 203>\n      model.compile(\"adam\", lr=0.001, external_trainable_variables=[m0,m1,m2,m3])\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\utils\\internal.py\", line 22, in wrapper\n      result = f(*args, **kwargs)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 124, in compile\n      self._compile_tensorflow_compat_v1(lr, loss_fn, decay, loss_weights)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 169, in _compile_tensorflow_compat_v1\n      losses_train = losses(self.data.losses_train)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 155, in losses\n      losses = losses_fn(\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\data.py\", line 13, in losses_train\n      return self.losses(targets, outputs, loss_fn, inputs, model, aux=aux)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\pde.py\", line 156, in losses\n      error = bc.error(self.train_x, inputs, outputs, beg, end)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\icbc\\boundary_conditions.py\", line 210, in error\n      return outputs[beg:end, self.component : self.component + 1] - self.values\nNode: 'sub_282'\nDetected at node 'sub_282' defined at (most recent call last):\n    File \"D:\\anaconda\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"D:\\anaconda\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel_launcher.py\", line 16, in <module>\n      app.launch_new_instance()\n    File \"D:\\anaconda\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 677, in start\n      self.io_loop.start()\n    File \"D:\\anaconda\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"D:\\anaconda\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 471, in dispatch_queue\n      await self.process_one()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 460, in process_one\n      await dispatch(*args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 367, in dispatch_shell\n      await result\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 662, in execute_request\n      reply_content = await reply_content\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 360, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 532, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2863, in run_cell\n      result = self._run_cell(\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2909, in _run_cell\n      return runner(coro)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n      coro.send(None)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3106, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3309, in run_ast_nodes\n      if await self.run_code(code, result, async_=asy):\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3369, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\86183\\AppData\\Local\\Temp\\ipykernel_19072\\2949448217.py\", line 203, in <cell line: 203>\n      model.compile(\"adam\", lr=0.001, external_trainable_variables=[m0,m1,m2,m3])\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\utils\\internal.py\", line 22, in wrapper\n      result = f(*args, **kwargs)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 124, in compile\n      self._compile_tensorflow_compat_v1(lr, loss_fn, decay, loss_weights)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 169, in _compile_tensorflow_compat_v1\n      losses_train = losses(self.data.losses_train)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 155, in losses\n      losses = losses_fn(\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\data.py\", line 13, in losses_train\n      return self.losses(targets, outputs, loss_fn, inputs, model, aux=aux)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\pde.py\", line 156, in losses\n      error = bc.error(self.train_x, inputs, outputs, beg, end)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\icbc\\boundary_conditions.py\", line 210, in error\n      return outputs[beg:end, self.component : self.component + 1] - self.values\nNode: 'sub_282'\n2 root error(s) found.\n  (0) INVALID_ARGUMENT: Incompatible shapes: [40000,1] vs. [200,1]\n\t [[{{node sub_282}}]]\n\t [[cluster_7_1/xla_compile]]\n\t [[cluster_7_1/merge_oidx_0/_1]]\n  (1) INVALID_ARGUMENT: Incompatible shapes: [40000,1] vs. [200,1]\n\t [[{{node sub_282}}]]\n\t [[cluster_7_1/xla_compile]]\n0 successful operations.\n0 derived errors ignored.\n\nOriginal stack trace for 'sub_282':\n  File \"D:\\anaconda\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"D:\\anaconda\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel_launcher.py\", line 16, in <module>\n    app.launch_new_instance()\n  File \"D:\\anaconda\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 677, in start\n    self.io_loop.start()\n  File \"D:\\anaconda\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"D:\\anaconda\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 471, in dispatch_queue\n    await self.process_one()\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 460, in process_one\n    await dispatch(*args)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 367, in dispatch_shell\n    await result\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 662, in execute_request\n    reply_content = await reply_content\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 360, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 532, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2863, in run_cell\n    result = self._run_cell(\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2909, in _run_cell\n    return runner(coro)\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n    coro.send(None)\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3106, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3309, in run_ast_nodes\n    if await self.run_code(code, result, async_=asy):\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3369, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\86183\\AppData\\Local\\Temp\\ipykernel_19072\\2949448217.py\", line 203, in <cell line: 203>\n    model.compile(\"adam\", lr=0.001, external_trainable_variables=[m0,m1,m2,m3])\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\utils\\internal.py\", line 22, in wrapper\n    result = f(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 124, in compile\n    self._compile_tensorflow_compat_v1(lr, loss_fn, decay, loss_weights)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 169, in _compile_tensorflow_compat_v1\n    losses_train = losses(self.data.losses_train)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 155, in losses\n    losses = losses_fn(\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\data.py\", line 13, in losses_train\n    return self.losses(targets, outputs, loss_fn, inputs, model, aux=aux)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\pde.py\", line 156, in losses\n    error = bc.error(self.train_x, inputs, outputs, beg, end)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\icbc\\boundary_conditions.py\", line 210, in error\n    return outputs[beg:end, self.component : self.component + 1] - self.values\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\util\\traceback_utils.py\", line 150, in error_handler\n    return fn(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\ops\\math_ops.py\", line 1407, in binary_op_wrapper\n    return func(x, y, name=name)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\util\\traceback_utils.py\", line 150, in error_handler\n    return fn(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\util\\dispatch.py\", line 1176, in op_dispatch_handler\n    return dispatch_target(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\ops\\math_ops.py\", line 548, in subtract\n    return gen_math_ops.sub(x, y, name)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\ops\\gen_math_ops.py\", line 11174, in sub\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mInvalidArgumentError\u001b[0m                      Traceback (most recent call last)",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\client\\session.py:1378\u001b[0m, in \u001b[0;36mBaseSession._do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1377\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 1378\u001b[0m   \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1379\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m errors\u001b[38;5;241m.\u001b[39mOpError \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\client\\session.py:1361\u001b[0m, in \u001b[0;36mBaseSession._do_run.<locals>._run_fn\u001b[1;34m(feed_dict, fetch_list, target_list, options, run_metadata)\u001b[0m\n\u001b[0;32m   1360\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_extend_graph()\n\u001b[1;32m-> 1361\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_tf_sessionrun\u001b[49m\u001b[43m(\u001b[49m\u001b[43moptions\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfeed_dict\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfetch_list\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1362\u001b[0m \u001b[43m                                \u001b[49m\u001b[43mtarget_list\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrun_metadata\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\client\\session.py:1454\u001b[0m, in \u001b[0;36mBaseSession._call_tf_sessionrun\u001b[1;34m(self, options, feed_dict, fetch_list, target_list, run_metadata)\u001b[0m\n\u001b[0;32m   1452\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_call_tf_sessionrun\u001b[39m(\u001b[38;5;28mself\u001b[39m, options, feed_dict, fetch_list, target_list,\n\u001b[0;32m   1453\u001b[0m                         run_metadata):\n\u001b[1;32m-> 1454\u001b[0m   \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mtf_session\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mTF_SessionRun_wrapper\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_session\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moptions\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfeed_dict\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1455\u001b[0m \u001b[43m                                          \u001b[49m\u001b[43mfetch_list\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtarget_list\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1456\u001b[0m \u001b[43m                                          \u001b[49m\u001b[43mrun_metadata\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[1;31mInvalidArgumentError\u001b[0m: 2 root error(s) found.\n  (0) INVALID_ARGUMENT: Incompatible shapes: [40000,1] vs. [200,1]\n\t [[{{node sub_282}}]]\n\t [[cluster_7_1/xla_compile]]\n\t [[cluster_7_1/merge_oidx_0/_1]]\n  (1) INVALID_ARGUMENT: Incompatible shapes: [40000,1] vs. [200,1]\n\t [[{{node sub_282}}]]\n\t [[cluster_7_1/xla_compile]]\n0 successful operations.\n0 derived errors ignored.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mInvalidArgumentError\u001b[0m                      Traceback (most recent call last)",
      "Input \u001b[1;32mIn [6]\u001b[0m, in \u001b[0;36m<cell line: 205>\u001b[1;34m()\u001b[0m\n\u001b[0;32m    203\u001b[0m model\u001b[38;5;241m.\u001b[39mcompile(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124madam\u001b[39m\u001b[38;5;124m\"\u001b[39m, lr\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0.001\u001b[39m, external_trainable_variables\u001b[38;5;241m=\u001b[39m[m0,m1,m2,m3])\n\u001b[0;32m    204\u001b[0m variable \u001b[38;5;241m=\u001b[39m dde\u001b[38;5;241m.\u001b[39mcallbacks\u001b[38;5;241m.\u001b[39mVariableValue([m0,m1,m2,m3], period\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1000\u001b[39m, filename\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mvariables.dat\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m--> 205\u001b[0m losshistory, train_state \u001b[38;5;241m=\u001b[39m \u001b[43mmodel\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtrain\u001b[49m\u001b[43m(\u001b[49m\u001b[43miterations\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m80000\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcallbacks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\u001b[43mvariable\u001b[49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    206\u001b[0m dde\u001b[38;5;241m.\u001b[39msaveplot(losshistory, train_state, issave\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m, isplot\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\utils\\internal.py:22\u001b[0m, in \u001b[0;36mtiming.<locals>.wrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m     19\u001b[0m \u001b[38;5;129m@wraps\u001b[39m(f)\n\u001b[0;32m     20\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapper\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m     21\u001b[0m     ts \u001b[38;5;241m=\u001b[39m timeit\u001b[38;5;241m.\u001b[39mdefault_timer()\n\u001b[1;32m---> 22\u001b[0m     result \u001b[38;5;241m=\u001b[39m f(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m     23\u001b[0m     te \u001b[38;5;241m=\u001b[39m timeit\u001b[38;5;241m.\u001b[39mdefault_timer()\n\u001b[0;32m     24\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m took \u001b[39m\u001b[38;5;132;01m%f\u001b[39;00m\u001b[38;5;124m s\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m (f\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m, te \u001b[38;5;241m-\u001b[39m ts))\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py:577\u001b[0m, in \u001b[0;36mModel.train\u001b[1;34m(self, iterations, batch_size, display_every, disregard_previous_best, callbacks, model_restore_path, model_save_path, epochs)\u001b[0m\n\u001b[0;32m    575\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39mset_data_train(\u001b[38;5;241m*\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdata\u001b[38;5;241m.\u001b[39mtrain_next_batch(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbatch_size))\n\u001b[0;32m    576\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39mset_data_test(\u001b[38;5;241m*\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdata\u001b[38;5;241m.\u001b[39mtest())\n\u001b[1;32m--> 577\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_test\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    578\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcallbacks\u001b[38;5;241m.\u001b[39mon_train_begin()\n\u001b[0;32m    579\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m optimizers\u001b[38;5;241m.\u001b[39mis_external_optimizer(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mopt_name):\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py:709\u001b[0m, in \u001b[0;36mModel._test\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    705\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_test\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m    706\u001b[0m     (\n\u001b[0;32m    707\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39my_pred_train,\n\u001b[0;32m    708\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39mloss_train,\n\u001b[1;32m--> 709\u001b[0m     ) \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_outputs_losses\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    710\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    711\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtrain_state\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mX_train\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    712\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtrain_state\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43my_train\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    713\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtrain_state\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtrain_aux_vars\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    714\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    715\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39my_pred_test, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39mloss_test \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_outputs_losses(\n\u001b[0;32m    716\u001b[0m         \u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    717\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39mX_test,\n\u001b[0;32m    718\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39my_test,\n\u001b[0;32m    719\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39mtest_aux_vars,\n\u001b[0;32m    720\u001b[0m     )\n\u001b[0;32m    722\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtrain_state\u001b[38;5;241m.\u001b[39my_test, (\u001b[38;5;28mlist\u001b[39m, \u001b[38;5;28mtuple\u001b[39m)):\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py:487\u001b[0m, in \u001b[0;36mModel._outputs_losses\u001b[1;34m(self, training, inputs, targets, auxiliary_vars)\u001b[0m\n\u001b[0;32m    485\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m backend_name \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtensorflow.compat.v1\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m    486\u001b[0m     feed_dict \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mnet\u001b[38;5;241m.\u001b[39mfeed_dict(training, inputs, targets, auxiliary_vars)\n\u001b[1;32m--> 487\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun\u001b[49m\u001b[43m(\u001b[49m\u001b[43moutputs_losses\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfeed_dict\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfeed_dict\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    488\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m backend_name \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtensorflow\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m    489\u001b[0m     outs \u001b[38;5;241m=\u001b[39m outputs_losses(inputs, targets, auxiliary_vars)\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\client\\session.py:968\u001b[0m, in \u001b[0;36mBaseSession.run\u001b[1;34m(self, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m    965\u001b[0m run_metadata_ptr \u001b[38;5;241m=\u001b[39m tf_session\u001b[38;5;241m.\u001b[39mTF_NewBuffer() \u001b[38;5;28;01mif\u001b[39;00m run_metadata \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    967\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 968\u001b[0m   result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_run\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfetches\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfeed_dict\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moptions_ptr\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    969\u001b[0m \u001b[43m                     \u001b[49m\u001b[43mrun_metadata_ptr\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    970\u001b[0m   \u001b[38;5;28;01mif\u001b[39;00m run_metadata:\n\u001b[0;32m    971\u001b[0m     proto_data \u001b[38;5;241m=\u001b[39m tf_session\u001b[38;5;241m.\u001b[39mTF_GetBuffer(run_metadata_ptr)\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\client\\session.py:1191\u001b[0m, in \u001b[0;36mBaseSession._run\u001b[1;34m(self, handle, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1188\u001b[0m \u001b[38;5;66;03m# We only want to really perform the run if fetches or targets are provided,\u001b[39;00m\n\u001b[0;32m   1189\u001b[0m \u001b[38;5;66;03m# or if the call is a partial run that specifies feeds.\u001b[39;00m\n\u001b[0;32m   1190\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m final_fetches \u001b[38;5;129;01mor\u001b[39;00m final_targets \u001b[38;5;129;01mor\u001b[39;00m (handle \u001b[38;5;129;01mand\u001b[39;00m feed_dict_tensor):\n\u001b[1;32m-> 1191\u001b[0m   results \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_do_run\u001b[49m\u001b[43m(\u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfinal_targets\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfinal_fetches\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1192\u001b[0m \u001b[43m                         \u001b[49m\u001b[43mfeed_dict_tensor\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moptions\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrun_metadata\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1193\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m   1194\u001b[0m   results \u001b[38;5;241m=\u001b[39m []\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\client\\session.py:1371\u001b[0m, in \u001b[0;36mBaseSession._do_run\u001b[1;34m(self, handle, target_list, fetch_list, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1368\u001b[0m   \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_call_tf_sessionprun(handle, feed_dict, fetch_list)\n\u001b[0;32m   1370\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m handle \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m-> 1371\u001b[0m   \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_do_call\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_run_fn\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfeeds\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfetches\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtargets\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moptions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1372\u001b[0m \u001b[43m                       \u001b[49m\u001b[43mrun_metadata\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1373\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m   1374\u001b[0m   \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_do_call(_prun_fn, handle, feeds, fetches)\n",
      "File \u001b[1;32m~\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\client\\session.py:1397\u001b[0m, in \u001b[0;36mBaseSession._do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1392\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124monly supports NHWC tensor format\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01min\u001b[39;00m message:\n\u001b[0;32m   1393\u001b[0m   message \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m (\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124mA possible workaround: Try disabling Grappler optimizer\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m   1394\u001b[0m               \u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124mby modifying the config for creating the session eg.\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m   1395\u001b[0m               \u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124msession_config.graph_options.rewrite_options.\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m   1396\u001b[0m               \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mdisable_meta_optimizer = True\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m-> 1397\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;28mtype\u001b[39m(e)(node_def, op, message)\n",
      "\u001b[1;31mInvalidArgumentError\u001b[0m: Graph execution error:\n\nDetected at node 'sub_282' defined at (most recent call last):\n    File \"D:\\anaconda\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"D:\\anaconda\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel_launcher.py\", line 16, in <module>\n      app.launch_new_instance()\n    File \"D:\\anaconda\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 677, in start\n      self.io_loop.start()\n    File \"D:\\anaconda\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"D:\\anaconda\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 471, in dispatch_queue\n      await self.process_one()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 460, in process_one\n      await dispatch(*args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 367, in dispatch_shell\n      await result\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 662, in execute_request\n      reply_content = await reply_content\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 360, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 532, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2863, in run_cell\n      result = self._run_cell(\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2909, in _run_cell\n      return runner(coro)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n      coro.send(None)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3106, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3309, in run_ast_nodes\n      if await self.run_code(code, result, async_=asy):\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3369, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\86183\\AppData\\Local\\Temp\\ipykernel_19072\\2949448217.py\", line 203, in <cell line: 203>\n      model.compile(\"adam\", lr=0.001, external_trainable_variables=[m0,m1,m2,m3])\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\utils\\internal.py\", line 22, in wrapper\n      result = f(*args, **kwargs)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 124, in compile\n      self._compile_tensorflow_compat_v1(lr, loss_fn, decay, loss_weights)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 169, in _compile_tensorflow_compat_v1\n      losses_train = losses(self.data.losses_train)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 155, in losses\n      losses = losses_fn(\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\data.py\", line 13, in losses_train\n      return self.losses(targets, outputs, loss_fn, inputs, model, aux=aux)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\pde.py\", line 156, in losses\n      error = bc.error(self.train_x, inputs, outputs, beg, end)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\icbc\\boundary_conditions.py\", line 210, in error\n      return outputs[beg:end, self.component : self.component + 1] - self.values\nNode: 'sub_282'\nDetected at node 'sub_282' defined at (most recent call last):\n    File \"D:\\anaconda\\lib\\runpy.py\", line 197, in _run_module_as_main\n      return _run_code(code, main_globals, None,\n    File \"D:\\anaconda\\lib\\runpy.py\", line 87, in _run_code\n      exec(code, run_globals)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel_launcher.py\", line 16, in <module>\n      app.launch_new_instance()\n    File \"D:\\anaconda\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n      app.start()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 677, in start\n      self.io_loop.start()\n    File \"D:\\anaconda\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n      self.asyncio_loop.run_forever()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n      self._run_once()\n    File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n      handle._run()\n    File \"D:\\anaconda\\lib\\asyncio\\events.py\", line 80, in _run\n      self._context.run(self._callback, *self._args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 471, in dispatch_queue\n      await self.process_one()\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 460, in process_one\n      await dispatch(*args)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 367, in dispatch_shell\n      await result\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 662, in execute_request\n      reply_content = await reply_content\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 360, in do_execute\n      res = shell.run_cell(code, store_history=store_history, silent=silent)\n    File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 532, in run_cell\n      return super().run_cell(*args, **kwargs)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2863, in run_cell\n      result = self._run_cell(\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2909, in _run_cell\n      return runner(coro)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n      coro.send(None)\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3106, in run_cell_async\n      has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3309, in run_ast_nodes\n      if await self.run_code(code, result, async_=asy):\n    File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3369, in run_code\n      exec(code_obj, self.user_global_ns, self.user_ns)\n    File \"C:\\Users\\86183\\AppData\\Local\\Temp\\ipykernel_19072\\2949448217.py\", line 203, in <cell line: 203>\n      model.compile(\"adam\", lr=0.001, external_trainable_variables=[m0,m1,m2,m3])\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\utils\\internal.py\", line 22, in wrapper\n      result = f(*args, **kwargs)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 124, in compile\n      self._compile_tensorflow_compat_v1(lr, loss_fn, decay, loss_weights)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 169, in _compile_tensorflow_compat_v1\n      losses_train = losses(self.data.losses_train)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 155, in losses\n      losses = losses_fn(\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\data.py\", line 13, in losses_train\n      return self.losses(targets, outputs, loss_fn, inputs, model, aux=aux)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\pde.py\", line 156, in losses\n      error = bc.error(self.train_x, inputs, outputs, beg, end)\n    File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\icbc\\boundary_conditions.py\", line 210, in error\n      return outputs[beg:end, self.component : self.component + 1] - self.values\nNode: 'sub_282'\n2 root error(s) found.\n  (0) INVALID_ARGUMENT: Incompatible shapes: [40000,1] vs. [200,1]\n\t [[{{node sub_282}}]]\n\t [[cluster_7_1/xla_compile]]\n\t [[cluster_7_1/merge_oidx_0/_1]]\n  (1) INVALID_ARGUMENT: Incompatible shapes: [40000,1] vs. [200,1]\n\t [[{{node sub_282}}]]\n\t [[cluster_7_1/xla_compile]]\n0 successful operations.\n0 derived errors ignored.\n\nOriginal stack trace for 'sub_282':\n  File \"D:\\anaconda\\lib\\runpy.py\", line 197, in _run_module_as_main\n    return _run_code(code, main_globals, None,\n  File \"D:\\anaconda\\lib\\runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel_launcher.py\", line 16, in <module>\n    app.launch_new_instance()\n  File \"D:\\anaconda\\lib\\site-packages\\traitlets\\config\\application.py\", line 846, in launch_instance\n    app.start()\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 677, in start\n    self.io_loop.start()\n  File \"D:\\anaconda\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 199, in start\n    self.asyncio_loop.run_forever()\n  File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 601, in run_forever\n    self._run_once()\n  File \"D:\\anaconda\\lib\\asyncio\\base_events.py\", line 1905, in _run_once\n    handle._run()\n  File \"D:\\anaconda\\lib\\asyncio\\events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 471, in dispatch_queue\n    await self.process_one()\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 460, in process_one\n    await dispatch(*args)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 367, in dispatch_shell\n    await result\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 662, in execute_request\n    reply_content = await reply_content\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 360, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"D:\\anaconda\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 532, in run_cell\n    return super().run_cell(*args, **kwargs)\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2863, in run_cell\n    result = self._run_cell(\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2909, in _run_cell\n    return runner(coro)\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n    coro.send(None)\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3106, in run_cell_async\n    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3309, in run_ast_nodes\n    if await self.run_code(code, result, async_=asy):\n  File \"D:\\anaconda\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3369, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"C:\\Users\\86183\\AppData\\Local\\Temp\\ipykernel_19072\\2949448217.py\", line 203, in <cell line: 203>\n    model.compile(\"adam\", lr=0.001, external_trainable_variables=[m0,m1,m2,m3])\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\utils\\internal.py\", line 22, in wrapper\n    result = f(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 124, in compile\n    self._compile_tensorflow_compat_v1(lr, loss_fn, decay, loss_weights)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 169, in _compile_tensorflow_compat_v1\n    losses_train = losses(self.data.losses_train)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\model.py\", line 155, in losses\n    losses = losses_fn(\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\data.py\", line 13, in losses_train\n    return self.losses(targets, outputs, loss_fn, inputs, model, aux=aux)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\data\\pde.py\", line 156, in losses\n    error = bc.error(self.train_x, inputs, outputs, beg, end)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\deepxde\\icbc\\boundary_conditions.py\", line 210, in error\n    return outputs[beg:end, self.component : self.component + 1] - self.values\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\util\\traceback_utils.py\", line 150, in error_handler\n    return fn(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\ops\\math_ops.py\", line 1407, in binary_op_wrapper\n    return func(x, y, name=name)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\util\\traceback_utils.py\", line 150, in error_handler\n    return fn(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\util\\dispatch.py\", line 1176, in op_dispatch_handler\n    return dispatch_target(*args, **kwargs)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\ops\\math_ops.py\", line 548, in subtract\n    return gen_math_ops.sub(x, y, name)\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\ops\\gen_math_ops.py\", line 11174, in sub\n    _, _, _op, _outputs = _op_def_library._apply_op_helper(\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\framework\\op_def_library.py\", line 797, in _apply_op_helper\n    op = g._create_op_internal(op_type_name, inputs, dtypes=None,\n  File \"C:\\Users\\86183\\AppData\\Roaming\\Python\\Python39\\site-packages\\tensorflow\\python\\framework\\ops.py\", line 3800, in _create_op_internal\n    ret = Operation(\n"
     ]
    }
   ],
   "source": [
    "\"\"\"Backend supported: tensorflow.compat.v1, tensorflow, pytorch, paddle\"\"\"\n",
    "import deepxde as dde\n",
    "import numpy as np\n",
    "import math\n",
    "\n",
    "\n",
    "def gen_traindata():\n",
    "\n",
    "    x_lower = -10\n",
    "    x_upper = 10\n",
    "    t_lower = -2\n",
    "    t_upper = 2\n",
    "    x = np.linspace(x_lower, x_upper, 200)\n",
    "    t = np.linspace(t_lower, t_upper, 200)\n",
    "    a=1\n",
    "    b=2\n",
    "    k_r=1.5\n",
    "    k_i=1\n",
    "    v1=1\n",
    "    v2=1\n",
    "    v3=1\n",
    "    c=math.log(a*a*v1+b*b*v2+a*b*v3+a*b*v3)/((2*k_r)*(2*k_r))\n",
    "    u1=(a/2) * np.exp(-c/2)*np.cos(x+1.25*t)/np.cosh(1.5*x-3*t+c/2)\n",
    "    v1=(a/2) * np.exp(-c/2)*np.sin(x+1.25*t)/np.cosh(1.5*x-3*t+c/2)\n",
    "    u2=(b/2) * np.exp(-c/2)*np.cos(x+1.25*t)/np.cosh(1.5*x-3*t+c/2)\n",
    "    v2=(b/2) * np.exp(-c/2)*np.sin(x+1.25*t)/np.cosh(1.5*x-3*t+c/2)\n",
    "\n",
    "    X, T = np.meshgrid(x, t)\n",
    "    X = np.reshape(X, (-1, 1))\n",
    "    T = np.reshape(T, (-1, 1))\n",
    "    U1=np.reshape(u1,(-1,1))\n",
    "    V1=np.reshape(v1,(-1,1))\n",
    "    U2=np.reshape(u2,(-1,1))\n",
    "    V2=np.reshape(v2,(-1,1))\n",
    "    \n",
    "    return np.hstack((X, T)), U1,V1,U2,V2\n",
    "\n",
    "\n",
    "#kf = dde.Variable(0.05)\n",
    "#D = dde.Variable(1.0)\n",
    "m0=dde.Variable(1.0)\n",
    "m1=dde.Variable(2.0)\n",
    "m2=dde.Variable(-1.0)\n",
    "m3=dde.Variable(0.0)\n",
    "\n",
    "def pde(x,y):\n",
    "    \n",
    "    u1 = y[:, 0:1]\n",
    "    v1 = y[:, 1:2]\n",
    "    u2 = y[:, 2:3]\n",
    "    v2 = y[:, 3:4]\n",
    "    \n",
    "\n",
    "    # 在'jacobian'中，i 是输出分量，j 是输入分量\n",
    "    u1_t = dde.grad.jacobian(y, x, i=0, j=1)\n",
    "    v1_t = dde.grad.jacobian(y, x, i=1, j=1)\n",
    "    u2_t = dde.grad.jacobian(y, x, i=2, j=1)\n",
    "    v2_t = dde.grad.jacobian(y, x, i=3, j=1)\n",
    "\n",
    "    u1_x = dde.grad.jacobian(y, x, i=0, j=0)\n",
    "    v1_x = dde.grad.jacobian(y, x, i=1, j=0)\n",
    "    u2_x = dde.grad.jacobian(y, x, i=2, j=0)\n",
    "    v2_x = dde.grad.jacobian(y, x, i=3, j=0)\n",
    "\n",
    "    # 在“hessian”中，i 和 j 都是输入分量。 （Hessian 原则上可以是 d^2y/dxdt、d^2y/d^2x 等）\n",
    "    # 输出组件由“组件”选择\n",
    "    u1_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)\n",
    "    v1_xx = dde.grad.hessian(y, x, component=1, i=0, j=0)\n",
    "    u2_xx = dde.grad.hessian(y, x, component=2, i=0, j=0)\n",
    "    v2_xx = dde.grad.hessian(y, x, component=3, i=0, j=0)\n",
    "\n",
    "    #f_u = u_t + 0.5 * v_xx + (u ** 2 + v ** 2) * v\n",
    "    #f_v = v_t - 0.5 * u_xx - (u ** 2 + v ** 2) * u\n",
    "    \n",
    "    f_u1=(\n",
    "    v1_t+u1_xx\n",
    "    +2*u1*(m0*(u1**2-v1**2)+m1*(u2**2-v2**2)+(m2+m3)*(u1*u2+v1*v2))\n",
    "    -2*v1*(2*m0*u1*v1+2*m1*u2*v2-(m2-m3)*(u1*v2-u2*v1))\n",
    "     )\n",
    "\n",
    "    f_v1=(\n",
    "    -u1_t+v1_xx\n",
    "    +2*v1*(m0*(u1**2-v1**2)+m1*(u2**2-v2**2)+(m2+m3)*(u1*u2+v1*v2))\n",
    "    +2*u1*(2*m0*u1*v1+2*m1*u2*v2-(m2-m3)*(u1*v2-u2*v1))\n",
    "    )\n",
    "\n",
    "    f_u2=(\n",
    "    v2_t+u2_xx\n",
    "    +2*u2*(m0*(u1**2-v1**2)+m1*(u2**2-v2**2)+(m2+m3)*(u1*u2+v1*v2))\n",
    "    -2*v2*(2*m0*u1*v1+2*m1*u2*v2-(m2-m3)*(u1*v2-u2*v1))\n",
    "    )\n",
    "\n",
    "    f_v2=(\n",
    "    -u2_t+v2_xx\n",
    "    +2*v2*(m0*(u1**2-v1**2)+m1*(u2**2-v2**2)+(m2+m3)*(u1*u2+v1*v2))\n",
    "    +2*u2*(2*m0*u1*v1+2*m1*u2*v2-(m2-m3)*(u1*v2-u2*v1))\n",
    "    )\n",
    "    return [f_u1, f_v1,f_u2,f_v2]\n",
    "\n",
    "\n",
    "def fun_bc(x):\n",
    "    return x[:, 0:1]\n",
    "\n",
    "\n",
    "#def fun_init(x):\n",
    "#    return np.exp(-20 * x[:, 0:1])\n",
    "\n",
    "def init_cond_u1(x):\n",
    "    \"2 sech(x)\"\n",
    "    return (1/2) * np.cos(1.5*x[:, 0:1]-(-0.587/2)) * np.exp(2.5-1.5*x[:, 0:1]) / np.cosh(1.5*x[:, 0:1]+(-0.587/2) )\n",
    "def init_cond_u2(x):\n",
    "    \"2 tanh(x)\"\n",
    "    return np.cos(1.5*x[:, 0:1]-(-0.587/2)) * np.exp(2.5-1.5*x[:, 0:1]) / np.cosh(1.5*x[:, 0:1]+(-0.587/2))\n",
    "\n",
    "def init_cond_v1(x):\n",
    "    return (1/2) * np.sin(1.5*x[:, 0:1]-(-0.587/2)) * np.exp(2.5-1.5*x[:, 0:1]) / np.cosh(1.5*x[:, 0:1]+(-0.587/2))\n",
    "\n",
    "def init_cond_v2(x):\n",
    "    return np.sin(1.5*x[:, 0:1]-(-0.587/2)) * np.exp(2.5-1.5*x[:, 0:1]) / np.cosh(1.5*x[:, 0:1]+(-0.587/2))\n",
    "\n",
    "\n",
    "#geom = dde.geometry.Interval(0, 1)\n",
    "#timedomain = dde.geometry.TimeDomain(0, 10)\n",
    "#geomtime = dde.geometry.GeometryXTime(geom, timedomain)\n",
    "\n",
    "x_lower = -10\n",
    "x_upper = 10\n",
    "t_lower = -2\n",
    "t_upper = 2\n",
    "\n",
    "# 创建 2D 域（用于绘图和输入）\n",
    "\n",
    "\n",
    "# 整个域变平\n",
    "#X_star = np.hstack((X.flatten()[:, None], T.flatten()[:, None]))\n",
    "\n",
    "# 空间和时间域/几何（对于 deepxde 模型）\n",
    "geom = dde.geometry.Interval(x_lower, x_upper)\n",
    "time_domain = dde.geometry.TimeDomain(t_lower, t_upper)\n",
    "geomtime = dde.geometry.GeometryXTime(geom, time_domain)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "bc_u1_0 = dde.icbc.DirichletBC(\n",
    "    geomtime,fun_bc, lambda _, on_boundary: on_boundary, component=0\n",
    ")\n",
    "bc_u1_1 = dde.icbc.DirichletBC(\n",
    "    geomtime, fun_bc, lambda _, on_boundary: on_boundary,  component=0\n",
    ")\n",
    "bc_v1_0 = dde.icbc.DirichletBC(\n",
    "    geomtime, fun_bc, lambda _, on_boundary: on_boundary, component=1\n",
    ")\n",
    "bc_v1_1 = dde.icbc.DirichletBC(\n",
    "    geomtime, fun_bc, lambda _, on_boundary: on_boundary, component=1\n",
    ")\n",
    "bc_u2_0 = dde.icbc.DirichletBC(\n",
    "    geomtime,fun_bc, lambda _, on_boundary: on_boundary,  component=2\n",
    ")\n",
    "bc_u2_1 = dde.icbc.DirichletBC(\n",
    "    geomtime, fun_bc, lambda _, on_boundary: on_boundary, component=2\n",
    ")\n",
    "bc_v2_0 = dde.icbc.DirichletBC(\n",
    "    geomtime,fun_bc, lambda _, on_boundary: on_boundary,component=3\n",
    ")\n",
    "bc_v2_1 = dde.icbc.DirichletBC(\n",
    "    geomtime, fun_bc, lambda _, on_boundary: on_boundary,  component=3\n",
    ")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#ic1 = dde.icbc.IC(geomtime, fun_init, lambda _, on_initial: on_initial, component=0)\n",
    "#ic2 = dde.icbc.IC(geomtime, fun_init, lambda _, on_initial: on_initial, component=1)\n",
    "\n",
    "ic_u1 = dde.icbc.IC(geomtime, init_cond_u1, lambda _, on_initial: on_initial, component=0)\n",
    "ic_v1 = dde.icbc.IC(geomtime, init_cond_v1, lambda _, on_initial: on_initial, component=1)\n",
    "ic_u2 = dde.icbc.IC(geomtime, init_cond_u2, lambda _, on_initial: on_initial, component=2)\n",
    "ic_v2 = dde.icbc.IC(geomtime, init_cond_v2, lambda _, on_initial: on_initial, component=3)\n",
    "\n",
    "observe_x, u1,v1,u2,v2 = gen_traindata()\n",
    "observe_y1 = dde.icbc.PointSetBC(observe_x, u1, component=0)\n",
    "observe_y2 = dde.icbc.PointSetBC(observe_x, v1, component=1)\n",
    "observe_y3 = dde.icbc.PointSetBC(observe_x, u2, component=2)\n",
    "observe_y4 = dde.icbc.PointSetBC(observe_x, v2, component=3)\n",
    "\n",
    "data = dde.data.TimePDE(\n",
    "    geomtime,\n",
    "    pde,\n",
    "    #[bc_a, bc_b, ic1, ic2, observe_y1, observe_y2],\n",
    "    [bc_u1_0, bc_u1_1, bc_v1_0, bc_v1_1,bc_u2_0, bc_u2_1, bc_v2_0, bc_v2_1, ic_u1, ic_v1,ic_u2, ic_v2,observe_y1, observe_y2,observe_y3,observe_y4],\n",
    "    num_domain=2000,\n",
    "    num_boundary=100,\n",
    "    num_initial=100,\n",
    "    anchors=observe_x,\n",
    "    num_test=50000,\n",
    ")\n",
    "net = dde.nn.FNN([2] + [20] * 3 + [4], \"silu\", \"Glorot uniform\")\n",
    "\n",
    "\n",
    "\n",
    "model = dde.Model(data, net)\n",
    "model.compile(\"adam\", lr=0.001, external_trainable_variables=[m0,m1,m2,m3])\n",
    "variable = dde.callbacks.VariableValue([m0,m1,m2,m3], period=1000, filename=\"variables.dat\")\n",
    "losshistory, train_state = model.train(iterations=80000, callbacks=[variable])\n",
    "dde.saveplot(losshistory, train_state, issave=True, isplot=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed21de72",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
